core/custom_field_api.php

```diff
@@ -788,24 +792,19 @@ function custom_field_get_id_from_name( $p_field_name ) {
 		return false;
 	}
 
-	if( isset( $g_cache_name_to_id_map[$p_field_name] ) ) {
-		return $g_cache_name_to_id_map[$p_field_name];
-	}
-
-	db_param_push();
-	$t_query = 'SELECT id FROM {custom_field} WHERE name=' . db_param();
-	$t_result = db_query( $t_query, array( $p_field_name ) );
-
-	$t_row = db_fetch_array( $t_result );
-
-	if( !$t_row ) {
-		$g_cache_name_to_id_map[$p_field_name] = false;
-		return false;
+	$p_field_name = mb_strtolower($p_field_name);
+	if( !isset( $g_cache_name_to_id_map[$p_field_name] ) ) {
+		# Build cache of lowercase custom fields names to id
+		if( !$g_cache_name_to_id_map ) {
+			$t_query = new DbQuery( "SELECT id, name FROM {custom_field}" );
+			foreach( $t_query->fetch_all() as $t_row ) {
+				$t_name = mb_strtolower($t_row['name']);
+				$g_cache_name_to_id_map[$t_name] = $t_row['id'];
+			}
+		}
 	}
 
-	$g_cache_name_to_id_map[$p_field_name] = $t_row['id'];
-
-	return $t_row['id'];
+	return $g_cache_name_to_id_map[$p_field_name] ?? false;
```

Comment:
You want to merge to master-2.25 which is the last version where we support PHP 5.
The null coalescing operator ?? is supported starting from PHP 7.0. https://www.php.net/manual/en/migration70.new-features.php

Reply:
Thanks for the review. You're absolutely right, good catch ! I'll fix that.